rpcserver.go

```diff
@@ -3435,6 +3436,46 @@ func handleSetGenerate(s *rpcServer, cmd interface{}, closeChan <-chan struct{})
 	return nil, nil
 }
 
+// Text used to signify that a signed message follows and to prevent
+// inadvertently signing a transaction.
+const messageSignatureHeader = "Bitcoin Signed Message:\n"
+
+// handleSignMessageWithPrivKey implements the signmessagewithprivkey command.
+func handleSignMessageWithPrivKey(s *rpcServer, cmd interface{}, closeChan <-chan struct{}) (interface{}, error) {
+	c := cmd.(*btcjson.SignMessageWithPrivKeyCmd)
+
+	wif, err := btcutil.DecodeWIF(c.PrivKey)
```

Comment:
`bitcoind` rejects WIFs that do not belong to the current network. Or so it seems. Here's an example for you to reproduce using a mainnet WIF:

```sh
$ ./btcctl --regtest --rpcuser=aaaa --rpcpass=bbbb signmessagewithprivkey 5HueCGU8rMjxEXxiPuD5BDku4MkFqeZyd4dZ1jvhTVqvbTLvyTJ iamsatoshi
G+brEqKGQgPCSkPJrBau7x32eFhT3Y5YXdEXKG2bg2E2Tqi0lU54t+rYrEezLcSDLS4TM+/AO8a3A9ybAfvyMLY=

$ bitcoin-cli -regtest signmessagewithprivkey 5HueCGU8rMjxEXxiPuD5BDku4MkFqeZyd4dZ1jvhTVqvbTLvyTJ iamsatoshi
error code: -5
error message:
Invalid private key
```

The fix would be to simply check `wif.IsForNet(s.cfg.ChainParams)`.

Reply:
Yes this was the case. Fixed by checking `wif.IsForNet(s.cfg.ChainParams)` as you suggested.